types/rdf-utils-dataset/index.d.ts

```diff
@@ -0,0 +1,13 @@
+// Type definitions for rdf-utils-dataset 1.1
+// Project: https://github.com/rdf-ext/rdf-utils-dataset
+// Definitions by: Chris Wilkinson <https://github.com/thewilkybarkid>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { DatasetIndexed } from 'rdf-dataset-indexed/dataset';
+import { BaseQuad, DatasetCore } from 'rdf-js';
+
+type OutQuad<T> = T extends DatasetCore<infer Q, any> ? Q : never;
+
+export function resource<D extends DatasetIndexed<BaseQuad>>(input: D, subject: OutQuad<D>['subject']): D;
```

Comment:
This is from module `./resource.js`

I think it should be reflected in the type exports too

Reply:
Also, this does not represent the actual CJS export. Should be

```
declare function resource()

export = resource
```

Reply:
Finally, the parameters

1. Why not make `subject: Quad_Subject` or even `subject: Term`?
2. Missing their param `cloned?: boolean`

Reply:
1. Either should work. `Term` feels a bit _too_ loose (`DefaultGraph` makes no sense here), so `Quad_Subject` could be the right level.
2. I didn't include this as it looks to be an implementation detail, rather than part of the API.

Reply:
Although, `BaseQuad` uses `Term` so it has to be that.